tokio/src/task/mod.rs

```diff
@@ -47,12 +52,12 @@ pub(crate) struct Task<S: 'static> {
     _p: PhantomData<S>,
 }
 
-unsafe impl<S: Send + Sync + 'static> Send for Task<S> {}
+unsafe impl<S: ScheduleSend + Send + Sync + 'static> Send for Task<S> {}
```

Comment:
The `+ Send + Sync` is redundant here I think?
Also, this whole bound is weird to me. It's not clear why the _scheduler_ needs to be `Send` for the `Task` to be, but that's probably just me missing something.

Reply:
I think this should just be `ScheduleSend + 'static`

Reply:
(@jonhoo for reference, _all_ schedulers used to be bounded by `Send + Sync`; I removed that bound and made it only required when the scheduler is `ScheduleSend`, but forgot to change it here)

Reply:
Is the `Send` bound even needed when the scheduler is `ScheduleSend`?